x-pack/platform/plugins/shared/streams_app/public/components/data_management/stream_detail_enrichment/processors/grok/grok_patterns_editor.tsx

```diff
@@ -205,9 +186,10 @@ const DraggablePatternInput = ({
             </EuiFlexItem>
             <EuiFlexItem>
               <Expression
-                draftGrokExpression={field.draftGrokExpression}
+                draftGrokExpression={draftGrokExpression}
                 grokCollection={grokCollection}
                 dataTestSubj="streamsAppPatternExpression"
+                onChange={onChange}
```

Comment:
Using the onChange handler prevents to create additional expression subscriptions and reassing the instance when something changes.